test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NetCorePackageReferenceProjectTests.cs

```diff
@@ -447,6 +456,1817 @@ public async Task GetInstalledVersion_WithAssetsFile_ChangingPackageSpec_Returns
             }
         }
 
+        [Fact]
```

Comment:
I don't know if you can do anything about this, NuGet.Client doesn't have good test infrastructure, or well designed classes for testing, in my opinion. But I just compared the first 2 tests, they're 149 and 131 lines long respectively, and contain less than 20 differences. This file has nearly 2000 new lines in 13 tests. I don't know how I'm supposed to review them.

Reply:
Currently I still don't know all the details of unit test tool here like `SimpleTestPackageContext` , `SimpleTestPackageUtility` etc.
I just saw previous unit test and tried to adapt to my need. I know there're repeated code, but they're not exactly same. Maybe next time I'll create something better than this. Some of unit tests tested here is maybe not important, I just tried to test all the possible complex cases.

Reply:
Also I more thought is I have fix for https://github.com/NuGet/Home/issues/9224 here too.
Most unit tests are for that one check for correctness, not for performance improvement.
I wish I created 2 separated PRs then things would be better.